planner/core/common_plans.go

```diff
@@ -206,6 +206,9 @@ func (e *Execute) getPhysicalPlan(ctx sessionctx.Context, is infoschema.InfoSche
 }
 
 func (e *Execute) rebuildRange(p Plan) error {
+	if p == nil {
```

Comment:
shall we rebuild plan if nil is returned from plan cache?

Reply:
No, we shall not. The purpose of the condition is to check a nil plan (i.e., ``Insert/Update/Delete.SelectPlan``) when calling ``rebuildRange`` recursively. In order to improve the readability, I'd like to relocate the check condition in the updated PR.